core/src/tools/bsmtp.cc

```diff
@@ -246,7 +246,7 @@ int main(int argc, char* argv[])
 #if defined(HAVE_WIN32)
   SOCKET s;
 #else
-  int s, r;
```

Comment:
that's a bogus fix.
We should move that down to line 455 where it is needed.
This would
1. reduce its scope
2. initialize it immediately
3. merge it into the other `#if defined(HAVE_WIN32)`

Reply:
Very good idea, however it does not work as in line 535 s is also used and this is out of the scope of line 455 .